[Feat] iOS 파트 1차 과제

cchanmi opened the pull request:
## 🌱 작업한 내용

- 로그인, 회원가입 화면 UI 구성
- 다른 UI 화면으로 데이터 전달하기
- 버튼을 눌러 textfield 값 지우기
- textfield에 값이 입력되면 로그인 버튼 활성화

## 🌱 PR Point

<!-- 피드백을 받고 싶은 부분이나, 공유하고 싶은 부분을 적어주세요. -->

- navigation bar back button을 주어진 이미지로 바꾸는 과정이 코드는 입력했지만 제대로 적용되지 않았습니다 ㅠㅠ 어떤 부분이 잘못되었을까요? (viewcontroller에 changeBackButton 함수입니다)
- 심화 과제는 못 했습니다. 구글링을 통해 코드를 따라치는 것보다 좀 더 생각해 보는 과정이 더 중요할 것 같아서 급한 마음으로 과제를 수행하지 않았습니다. 심화 과제를 어떤 방식으로 구현하면 좋을지 방향을 조금 알려 주시면 참고할 수 있을 것 같습니다 🙏 🙏
- 깃이 아직 익숙하지 않아서 😂😂 공부를 더 해서 다음번에는 좀 더 깔끔하게 올려 볼게요 . . .

## 📸 스크린샷

<!-- 작업한 화면이 있다면 스크린 샷으로 첨부해주세요. -->

|    구현 내용    |   스크린샷   |
| :-------------: | :----------: |
| 로그인 화면 |![Simulator Screen Recording - iPhone 13 Pro - 2022-04-10 at 15 37 09](https://user-images.githubusercontent.com/86944161/162605738-b512ded1-c7ac-4ebd-8e08-251f14983f11.gif) 
| 가입 화면 |![Simulator Screen Recording - iPhone 13 Pro - 2022-04-10 at 15 37 09](https://user-images.githubusercontent.com/86944161/162605800-585b4fa1-b3f4-4d88-a0a4-e5b2447c4b18.gif)
|textfield 버튼 눌러 지우기|![Simulator Screen Recording - iPhone 13 Pro - 2022-04-10 at 15 41 59](https://user-images.githubusercontent.com/86944161/162605850-3e544ade-5616-4e89-a1fc-ef9c80dcdaef.gif)

## 📮 관련 이슈

<!-- 작업한 이슈번호를 # 뒤에 붙여주세요. -->

- Resolved: #2

Commit:
1차 과제 업로드

Commit:
데이터 전달 수정 부분

hellozo0 commented on instagramLogin/instagramLogin/PasswordViewController.swift:
```diff
@@ -0,0 +1,47 @@
+    // 다음 버튼, 입력받은 데이터를 다음 VC를 modal로 present 함.
+    @IBAction func nextButton(_ sender: Any) {
+        
+        guard let segue = self.storyboard?.instantiateViewController(withIdentifier: "CompletionViewController") as? CompletionViewController else { return }
```
여기에 `segue` 보다는 `CompletionViewController`에 대한 것이니 `completionVC`라고 하는게 좀 더 나중에 안 헷갈릴거 같아욤~!

cchanmi replied:
다른 변수들은 다 비슷하게 설정했는데 여기서 갑자기 segue가 튀어나와버렸네요 😂😂
수정했습니다 감사합니다~~!  🙏

hellozo0 commented on instagramLogin/instagramLogin/PasswordViewController.swift:
```diff
@@ -0,0 +1,47 @@
+class PasswordViewController: UIViewController {
+    
+    var nameData: String?
+    @IBOutlet weak var buttonState: UIButton!
```
요것도 나중에 `buttonstate` 활용할 경우가 생길때 이게 버튼인지 아닌지 파악하기가 어려워서 IBOutlet 요소들은 `다음으로 이동하는 Button`이면 `nextBtn`이나 `nextButton` 이라고 해주는게 보기도 편하고 어떤 것인지 파악하기가 쉽답니다..!

hellozo0 commented on instagramLogin/instagramLogin/PasswordViewController.swift:
```diff
@@ -0,0 +1,47 @@
+    
+    var nameData: String?
+    @IBOutlet weak var buttonState: UIButton!
+    @IBOutlet weak var textField: UITextField!
```
애도 조금 더 어떤 용도의 textfield인지 추가하면 좋을거 같아요! 비밀번호 받는 textfield이니까 `passwordTextField` 도 좋고 `pwTextField` 도 좋을거 가타요!

hellozo0 commented on instagramLogin/instagramLogin/ViewController.swift:
```diff
@@ -0,0 +1,69 @@
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        imgView.image = UIImage(named: "Instagram Black Logo.png")
+        userName.clearButtonMode = .whileEditing
```
사실... viewDidLoad()에는 코드를? 적는것이 좋지 않다고 해요...! 그래서 대부분 viewDidLoad에서는 함수들만 호출하는데요 그래서 위의 2줄은 따로 func 으로 빼고 viewdidLoad()에는 함수만 호출하는 식으로 하는게 좋을거 같아요!

```
func setUI() {
    imgView.image = UIImage(named: "Instagram Black Logo.png")
    userName.clearButtonMode = .whileEditing
}
```

hellozo0 reviewed (approved):
왕! 넘 고생했습니다...촤하하하......
그리고 생각하는 것도 좋지만 처음에는 어떤 프로퍼티? 기능들이 있는지 잘 모르니 구글링을 해보고 빨리 배우고 적용해보는 것도 나쁘진 않은거 같아요! 저도 완젼 처음이였어서 혼자 아무리 해도 뭘 해야할지 모르겠더라구요... 그래서 좀 생각할 시간이 필요하다 싶으면 공식 문서에서 어떤 기능들이 있는지 찾아보고 혼자 뚱땅뚱땅 작업하다가 나중에 back버튼 이미지 바꾸는법 등등 이런식으로 정답을 찾아보는것도 좋은거 같습니다!!




그 backbutton 을 주어진 이미지로 적용하는 방법은 찬미님의 `ViewController`파일에서 가 아니라 <버튼이 보여지는 `PasswordViewController` 파일과 `UsernameViewController` 파일에서 해당 코드들을 작성해줘야해요! 
아래의 코드들을 각 2개의 뷰컨트롤러 파일에 추가 해주고 `viewdidLoad`에 `changeBackButton()`을 적어주기만 하면 적용이 되서 보일 거에요!!
```
private func changeBackButton() {
    let backButton = UIBarButtonItem(image: UIImage(named: "icn_back_24"), style: .done, target: self, action: #selector(back))
    self.navigationItem.leftBarButtonItem = backButton
  }
```

```
  @objc func back() {
    self.navigationController?.popViewController(animated: true)
  }
```

ffalswo2 commented on instagramLogin/instagramLogin/ViewController.swift:
```diff
@@ -0,0 +1,69 @@
+    
+    // 가입하기 버튼, 다음 VC로 데이터 push
+    @IBAction func joinButton(_ sender: Any) {
+        guard let nextVC = self.storyboard?.instantiateViewController(withIdentifier: "UsernameViewController") as? UsernameViewController else { return }
```
이거는 제 개인적인 생각일 수도 있어서 다른분의 의견도 궁금한데 한번 공유해볼게요?!
nextVC라고 했을때 나중에 뷰컨이 많아지게 되면 봤을때 next가 어디 뷰컨을 말하는건지 한눈에 봤을때 직관적이지 않은 것 같아서 이런 변수 네이밍 할 때 구체적으로 어떤 뷰 컨트롤러인지 좀 더 자세히 적는 것도 좋은 방법일 것 같습니다!

cchanmi replied:
세미나 실습 때 입력했던 변수를 습관적으로 다시 사용하였네요...!
좋은 피드백 감사합니다 😇

jimin-kiim commented on instagramLogin/instagramLogin/PasswordViewController.swift:
```diff
@@ -0,0 +1,47 @@
+    }
+    
+    private func setButton() {
+        if textField.hasText {
```
코드 컨벤션 자료에서 if 문 쓸 때 조건문 괄호 치지 말기 내용을 보고 지나쳤던 것 같은데 찬미 님 코드를 보고나니 기억이 나네요! 컨벤션 지켜서 잘 써주신 것 같아요 😆👍🏻  제가 보기엔 너무 잘 해주신 것 같네요ㅎㅎ 공부를 더 열심히 해서 피드백할 내용을 알아볼 줄 아는 팀원이 되겠습니다ㅎㅎ 과제 수고하셨어요!!

cchanmi replied:
지민님도 과제 고생하셨습니다!
저도 많이 부족한 상태라 앞으로 더 열심히 공부해서 든든한 팀원이 되고 싶네요
같이 파이팅합시다 ~~ 👏

Commit:
navigation bar back button 바뀌지 않는 것 해결

Commit:
Button 아웃렛변수명 좀 더 파악하기 쉽게 수정

lsj8706 commented on instagramLogin/instagramLogin/CompletionViewController.swift:
```diff
@@ -0,0 +1,36 @@
+        guard let firstVC = self.presentingViewController as? UINavigationController else { return }
+
+        self.dismiss(animated: true) {
+            firstVC.popToRootViewController(animated: true)
```
저도 처음에 이렇게 dismiss의 completion에서 popToRootViewController()를 사용했었는데, 이렇게 하면 바로 처음 화면으로 가지지 않고 아마 패스워드 입력받는 화면이 잠시 보였다가 처음 화면으로 가질거에요. 만약 완료하기 버튼을 눌렀을때 바로 로그인화면(처음화면)으로 가게 하고 싶다면 여기에 
```
            firstVC.popToRootViewController(animated: true)
```
이 코드를 지우고 PasswordViewController에 있는 present를
```
            present(segue, animated: true) {
            self.navigationController?.popToRootViewController(animated: true)
         }
```
이렇게 수정해보세요!

Commit:
textField 변수명 좀 더 파악하기 쉽게 수정

Commit:
✅[CHORE] : 코드 수정, 내부 파일 수정 (#2)

##### 제목은 최대 50 글자까지만 입력 ############## -> |
# 본문은 위에 작성
######## 본문은 한 줄에 최대 72 글자까지만 입력 ########################### -> |
# 꼬릿말은 아래에 작성: ex) #이슈 번호
# --- COMMIT END ---
#- ✅[CHORE] : 코드 수정, 내부 파일 수정
#- ✨[FEAT] : 새로운 기능 구현
#- ➕[ADD] : Feat 이외의 부수적인 코드 추가, 라이브러리 추가, 새로운 파일 생성 시
#- 🍱[ADD] : 에셋 추가
#- 🚑️[HOTFIX] : issue나, QA에서 급한 버그 수정에 사용
#- 🔨[FIX] : 버그, 오류 해결
#- ⚰️ [DEL] : 쓸모없는 코드 삭제
#- 📝[DOCS] : README나 WIKI 등의 문서 개정
#- 💄[MOD] : storyboard 파일,UI 수정한 경우
#- ✏️ [CORRECT] : 주로 문법의 오류나 타입의 변경, 이름 변경 등에 사용합니다.
#- 🚚[MOVE] : 프로젝트 내 파일이나 코드의 이동
#- ⏪️[RENAME] : 파일 이름 변경이 있을 때 사용합니다.
#- ⚡️[IMPROVE] : 향상이 있을 때 사용합니다.
#- ♻️[[REFACTOR] : 전면 수정이 있을 때 사용합니다
#- 🔀[MERGE]: 다른브렌치를 merge 할 때 사용합니다.
# ------------------
#     제목 첫 글자를 대문자로
#     제목은 명령문으로
#     제목 끝에 마침표(.) 금지
#     제목과 본문을 한 줄 띄워 분리하기
#     본문은 "어떻게" 보다 "무엇을", "왜"를 설명한다.
#     본문에 여러줄의 메시지를 작성할 땐 "-"로 구분
# ------------------

Commit:
Merge remote-tracking branch 'origin/assignment1' into assignment1

EunHee-Jeong commented on instagramLogin/instagramLogin/CompletionViewController.swift:
```diff
@@ -0,0 +1,36 @@
+    private func setNameData() {
+        if let nameData = nameData {
+            // 전닯받은 데이터 + 문장을 합쳐서 문자열을 출력한다
+            dataLabel.text = nameData + "님, instagram에 오신 것을 환영합니다."
```
문자열 보간법을 사용할 수도 있을 것 같아요 !

`dataLabel.text = "\(nameData)님, instagram에 오신 것을 환영합니다."`

cchanmi replied:
수정했습니다~~!

EunHee-Jeong commented on instagramLogin/instagramLogin/UsernameViewController.swift:
```diff
@@ -0,0 +1,54 @@
+    
+    
+    // 다음 버튼, textfield의 데이터를 다음 VC로 넘겨 준다. push
+    @IBAction func nextButton(_ sender: Any) {
+        guard let nextVC = self.storyboard?.instantiateViewController(withIdentifier: "PasswordViewController") as? PasswordViewController else { return }
+        
+        nextVC.nameData = nameTextField.text
+        
+        self.navigationController?.pushViewController(nextVC, animated: true)
+    }
+
+
+    @IBAction func nameEdit(_ sender: Any) {
+        setButton()
+    }
```
컨벤션) IBAction 같은 경우에는 프로퍼티(IBOutlet)가 아닌 동작을 정의해주는 부분이라서, 명사+동사 이런 식으로 해주는게 좋아요 !

네이밍은 정말 취향적인 부분이기 때문에,,, 첫 번째만 예시로 바꿔보자면, `touchUpToNextButton` 이나 `nextButtonDidTapped` 등이 있겠네요 ㅎㅎ

EunHee-Jeong commented on instagramLogin/instagramLogin/ViewController.swift:
```diff
@@ -0,0 +1,64 @@
+        setButton()
+    }
+    
+    private func setUI() {
+        imgView.image = UIImage(named: "Instagram Black Logo.png")
+        nameTextfield.clearButtonMode = .whileEditing
+    }
```
얘네 코드 말고 인스펙터 창에서 GUI 방식으로 넣어줄 수 있어요 ! 지민님 리뷰에 달았긴한데 여기도 예시 첨부할게요

![image](https://user-images.githubusercontent.com/70689381/162797880-8fc8e202-f156-4887-b996-25c46613f4e0.png)

EunHee-Jeong reviewed (commented):
과제하시느라 수고 많으셨습니다 !
노션 금잔디 페이지에 올려드린 깃 플로우 전략 한 번 읽어보시구 적용해서 리뷰 반영한 코드 커밋하시면 좋을 것 같아요 ㅎㅎ

Commit:
✨[FEAT] : 심화 과제 구현 #2

Commit:
✨[FEAT] : 심화 과제 구현 #2

##### 제목은 최대 50 글자까지만 입력 ############## -> |
# 본문은 위에 작성
######## 본문은 한 줄에 최대 72 글자까지만 입력 ########################### -> |
# 꼬릿말은 아래에 작성: ex) #이슈 번호
# --- COMMIT END ---
#- ✅[CHORE] : 코드 수정, 내부 파일 수정
#- ✨[FEAT] : 새로운 기능 구현
#- ➕[ADD] : Feat 이외의 부수적인 코드 추가, 라이브러리 추가, 새로운 파일 생성 시
#- 🍱[ADD] : 에셋 추가
#- 🚑️[HOTFIX] : issue나, QA에서 급한 버그 수정에 사용
#- 🔨[FIX] : 버그, 오류 해결
#- ⚰️ [DEL] : 쓸모없는 코드 삭제
#- 📝[DOCS] : README나 WIKI 등의 문서 개정
#- 💄[MOD] : storyboard 파일,UI 수정한 경우
#- ✏️ [CORRECT] : 주로 문법의 오류나 타입의 변경, 이름 변경 등에 사용합니다.
#- 🚚[MOVE] : 프로젝트 내 파일이나 코드의 이동
#- ⏪️[RENAME] : 파일 이름 변경이 있을 때 사용합니다.
#- ⚡️[IMPROVE] : 향상이 있을 때 사용합니다.
#- ♻️[[REFACTOR] : 전면 수정이 있을 때 사용합니다
#- 🔀[MERGE]: 다른브렌치를 merge 할 때 사용합니다.
# ------------------
#     제목 첫 글자를 대문자로
#     제목은 명령문으로
#     제목 끝에 마침표(.) 금지
#     제목과 본문을 한 줄 띄워 분리하기
#     본문은 "어떻게" 보다 "무엇을", "왜"를 설명한다.
#     본문에 여러줄의 메시지를 작성할 땐 "-"로 구분
# ------------------

Commit:
Merge remote-tracking branch 'origin/assignment1' into assignment1
